roles/load-balancers/manage-haproxy/handlers/main.yml

```diff
@@ -7,21 +7,30 @@
     state: started
   with_items:
   - haproxy
+  become: True
 
 - name: 'restart rsyslog'
   service:
     name: rsyslog
     state: restarted
+  become: True
 
 - name: 'reload haproxy'
-  service: 
+  service:
     name: haproxy
-    state: reloaded 
+    state: reloaded
+  become: True
 
 - name: 'remove tmp new file'
   file:
     path: '{{ temp_new_file }}'
     state: absent
+  delegate_to: localhost
+  run_once: True
 
-    
-
+- name: 'cleanup temp dir'
+  file:
+    path: '{{ haproxy_temp_dir }}'
+    state: absent
+  delegate_to: localhost
+  run_once: True
```

Comment:
We need to add become: True here as well otherwise we'll get another permission denied error.

Reply:
Looking back at this, I think we may be good without `become: true` here. I had assumed that the `haproxy_temp_dir` was referring to a dir that the temp file had been stored in. It looks like this is actually referring to the temp dirs that are created under `tmp`, which don't get owned as root.

Reply:
yup - this is all local and owned by the caller, so no need to change this.